src/coreclr/src/System.Private.CoreLib/src/System/GC.cs

```diff
@@ -54,28 +54,20 @@ public enum GCNotificationStatus
     public static class GC
     {
         [MethodImpl(MethodImplOptions.InternalCall)]
-        internal static extern void GetMemoryInfo(out ulong highMemLoadThresholdBytes,
-                                                  out ulong totalAvailableMemoryBytes,
-                                                  out ulong lastRecordedMemLoadBytes,
-                                                  out uint lastRecordedMemLoadPct,
-                                                  // The next two are size_t
-                                                  out UIntPtr lastRecordedHeapSizeBytes,
-                                                  out UIntPtr lastRecordedFragmentationBytes);
+        private static extern void GetMemoryInfo(GCMemoryInfoData data, int kind);
 
         public static GCMemoryInfo GetGCMemoryInfo()
         {
-            GetMemoryInfo(out ulong highMemLoadThresholdBytes,
-                          out ulong totalAvailableMemoryBytes,
-                          out ulong lastRecordedMemLoadBytes,
-                          out uint _,
-                          out UIntPtr lastRecordedHeapSizeBytes,
-                          out UIntPtr lastRecordedFragmentationBytes);
+            var data = new GCMemoryInfoData();
+            GetMemoryInfo(data, -1);
+            return new GCMemoryInfo(data);
+        }
 
-            return new GCMemoryInfo(highMemoryLoadThresholdBytes: (long)highMemLoadThresholdBytes,
-                                    memoryLoadBytes: (long)lastRecordedMemLoadBytes,
-                                    totalAvailableMemoryBytes: (long)totalAvailableMemoryBytes,
-                                    heapSizeBytes: (long)(ulong)lastRecordedHeapSizeBytes,
-                                    fragmentedBytes: (long)(ulong)lastRecordedFragmentationBytes);
+        public static GCMemoryInfo GetGCMemoryInfo(GCKind kind)
+        {
+            var data = new GCMemoryInfoData();
+            GetMemoryInfo(data, (int)kind);
```

Comment:
-1 ends up being valid.  Should it be in the enum?

Reply:
I thought about making it part of the enum but since this was the API surface we agreed on I didn't change it. should I make it part of the enum? I'm fine either way.

Reply:
What does -1 end up meaning... "All"?

Reply:
-1 means any. change to this? 
```csharp
    public enum GCKind
    {
        Ephemeral = 0,    // gen0 or gen1 GC
        FullBlocking = 1, // blocking gen2 GC
        Background = 2, // background GC (always gen2)
        Any = 3               // any kind
    };
```

Reply:
I do think we should add it to the enum, and Any sounds like a fine name, though I'm not sure about the numerical value.

@terrajobst, @bartonjs, opinions?

Reply:
just to clarify - Ephmeral/FullBlocking/Background are all exclusive - a GC can only be one of these 3 kinds. and Any means any **one** of these 3 kinds.

Reply:
Understood.  My question about the value is when in the future we introduce a new kind and it gets value 4, it's a little strange that any is in the middle.

Reply:
I see; it could be a much larger number like 0xff?

Reply:
`0` would be the most natural choice.

Reply:
0 SGTM.